src/plugins/discover/public/application/angular/discover.js

```diff
@@ -264,7 +264,8 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     if (!_.isEqual(newStatePartial, oldStatePartial)) {
       $scope.$evalAsync(async () => {
         if (oldStatePartial.index !== newStatePartial.index) {
-          //in case of index switch the route has currently to be reloaded, legacy
+          //in case of index pattern switch the route has currently to be reloaded, legacy
+          $route.reload();
```

Comment:
when there is a change of index pattern in URL, the route is reloaded, so the new index pattern is selected ... reload is necessary because of other legacy dependencies, and will be no longer necessary in the future, when moving to react router